### ISSUE TITLE
Go Agent installation from RPM/yum fails the first time

### ISSUE THREAD

**Issue by akshaydewan** [gocd/gocd#6947]
##### Issue Type
- Bug Report

##### Summary
Installing the Go Agent using RPM or yum on centos7 fails the first time. It completes successfully when the same command is run second time. The first failure shows
```
useradd: cannot create directory /var/go
```
I faced this twice, once using `yum` on 19.7 and once with `rpm -i` on 19.8

##### Environment
CentOS 7
Go Agent Version: 19.7 and 19.8

##### Console output
```
[vagrant@localhost ~]$ sudo rpm -ivh go-agent-19.7.0-9567.noarch.rpm
warning: go-agent-19.7.0-9567.noarch.rpm: Header V4 RSA/SHA1 Signature, key ID 8816c449: NOKEY
Preparing...                          ################################# [100%]
useradd: cannot create directory /var/go
error: %pre(go-agent-19.7.0-9567.noarch) scriptlet failed, exit status 12
error: go-agent-19.7.0-9567.noarch: install failed
[vagrant@localhost ~]$ sudo rpm -ivh go-agent-19.7.0-9567.noarch.rpm
warning: go-agent-19.7.0-9567.noarch.rpm: Header V4 RSA/SHA1 Signature, key ID 8816c449: NOKEY
Preparing...                          ################################# [100%]
Updating / installing...
   1:go-agent-19.7.0-9567             ################################# [100%]
Detected RHEL or Fedora:
Installing the go-agent daemon using systemd...
creating default service file...
Created symlink from /etc/systemd/system/multi-user.target.wants/go-agent.service to /etc/systemd/system/go-agent.service.
Installation of GoCD Agent completed.
Now please edit /usr/share/go-agent/wrapper-config/wrapper-properties.conf and set the server to the url address of your GoCD Server (https://example.com:8154/go).
To ensure full end-to-end transport security between the agent and server
see https://docs.gocd.org/current/installation/ssl_tls/end_to_end_transport_security.html
Once that is done start the GoCD Agent with 'systemctl start go-agent'
```

**Comment by rajiesh** [comment]
Looks like a Vagrant specific issue. Could reproduce on Centos-7 Vagrant box. Not reproducible on docker containers

**Comment by ketan** [comment]
I suspect it's more to do with selinux.

**Comment by ketan** [comment]
https://github.com/gocd/gocd/issues/4011, I think may be related.

**Comment by akshaydewan** [comment]
Yup, with `sudo setenforce 0` before the installation, it works in the first try. So seems selinux related. 

It's a bit crazy that the installation goes through the in next attempt.

Closing as duplicate of #4011
